### PR TITLE
Feature/city list

### PR DESCRIPTION
**Pull request by alen3003**

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Uncomment the next line to define a global platform for your project
+# platform :ios, '9.0'
+
+target 'WeatherPlannerDemo' do
+  # Comment the next line if you don't want to use dynamic frameworks
+  use_frameworks!
+
+  # Pods for WeatherPlannerDemo
+pod 'PureLayout'
+pod 'SwiftLint'
```
Comment (labasbozidar) [inline Podfile:10]: 👍

File: WeatherPlannerDemo/AppDelegate.swift [hunk 2]
```diff
@@ -3,28 +3,18 @@ import UIKit
...
-                                                 language: WeatherResponseLanguage.hr.rawValue).propertyPairs()
-        networkModuleClient.get(path: ApiEndpoints.weather.rawValue, queryParameters: cityParameters, memberType: City.self) { (result) in
-            switch result{
-            case .success(let city):
-                print(city.name)
-            case .failure(_):
-                break
-            }
-        }
+    func application(_ application: UIApplication,
```
Comment (labasbozidar) [inline WeatherPlannerDemo/AppDelegate.swift]:
```
func application(
   _ application: UIApplication,
  didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
) -> Bool {
   ...
}
```

We are using [this style](https://github.com/fiveagency/ios-swift-guide/tree/master/Code%20Style#function-declarations) for function declaration. Refactor this in other functions also.

File: WeatherPlannerDemo/Localization/LocalizationService.swift [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    }
+}
+
+extension LocalizationKey {
+    static func getLocalizationKey(for string: String) -> LocalizationKey? {
+        return LocalizationKey(rawValue: string)
+    }
+}
+
+extension String {
```
Comment (labasbozidar) [inline WeatherPlannerDemo/Localization/LocalizationService.swift:17]: Move this extension into separate file.

File: WeatherPlannerDemo/Network/BaseApiClient.swift [hunk 4]
```diff
@@ -12,7 +12,7 @@ public class BaseApiClient: ApiClientProtocol {
     
     public func get<ResultType: Decodable>(
         path: String,
-        queryParameters: [String : String]?,
+        queryParameters: [String: String]?,
```
Comment (labasbozidar) [inline WeatherPlannerDemo/Network/BaseApiClient.swift:15]: swiftLint is doing its job well 👁️

File: WeatherPlannerDemo/Network/QueryParameters/CitiesInCircleParameters.swift [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+struct CitiesInCircleParameters: QueryParameters, CustomReflectable {
+    let appid: String = ApiKey.main.rawValue
+    let latitude: String
+    let longitude: String
+    let radius: String
+    var unitSystem: String = MeasurementUnit.metric.rawValue
+    var language: String = WeatherResponseLanguage.en.rawValue
+    
+    var customMirror: Mirror {
+
```
Comment (labasbozidar) [inline WeatherPlannerDemo/Network/QueryParameters/CitiesInCircleParameters.swift]: You can remove this empty line.

File: WeatherPlannerDemo/UI/Components/CityListTableViewCell+Design.swift [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        
+        descriptionLabel = UILabel()
+        cityNameLabel = UILabel()
+        temperatureLabel = UILabel()
+        
+        innerStackView = UIStackView(arrangedSubviews: [descriptionLabel, cityNameLabel])
+        stackView = UIStackView(arrangedSubviews: [innerStackView, temperatureLabel])
+        
+        contentView.addSubview(stackView)
+
```
Comment (labasbozidar) [inline WeatherPlannerDemo/UI/Components/CityListTableViewCell+Design.swift]: Empty line.

File: WeatherPlannerDemo/UI/Components/CityListTableViewCell+Design.swift [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        selectionStyle = .none
+        
+        backgroundImageView.layer.masksToBounds = true
+        backgroundImageView.layer.cornerRadius = 8
+        
+        descriptionLabel.textColor = .white
+        cityNameLabel.textColor = .white
+        temperatureLabel.textColor = .white
+        
+        descriptionLabel.font = .boldSystemFont(ofSize: 40)
```
Comment (labasbozidar) [inline WeatherPlannerDemo/UI/Components/CityListTableViewCell+Design.swift]: You can create some class for fonts. Usually, we have predefined fonts.

File: WeatherPlannerDemo/UI/Components/CityListTableViewCell.swift [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    
+    static let reuseIdentifier = String(describing: self)
+    
+    var backgroundImageView: UIImageView!
+    var descriptionLabel: UILabel!
+    var cityNameLabel: UILabel!
+    var temperatureLabel: UILabel!
+    var innerStackView: UIStackView!
+    var stackView: UIStackView!
+
```
Comment (labasbozidar) [inline WeatherPlannerDemo/UI/Components/CityListTableViewCell.swift]: Empty line.

File: WeatherPlannerDemo/UI/ViewModelLists/CityListViewModelList.swift [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+            latitude: "\(coordinate.latitude)",
+            longitude: "\(coordinate.longitude)",
+            radius: "\(citiesInRange)",
+            language: WeatherResponseLanguage.hr.rawValue
+        ).propertyPairs()
+        
+        networkClient.get(path: ApiEndpoints.citiesInCircle.rawValue,
+                          queryParameters: parameters,
+                          memberType: CitiesInCircle.self
+        ) { [weak self] (result) in
```
Comment (labasbozidar) [inline WeatherPlannerDemo/UI/ViewModelLists/CityListViewModelList.swift:38]: You can use:
```
guard let self = self else { return }
```

File: WeatherPlannerDemo/UI/ViewModels/CityViewModel.swift [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    var minTemperature: String {
+        return "\(Int(city.main.temp_min))°"
+    }
+    
+    var maxTemperature: String {
+        return "\(Int(city.main.temp_max))°"
+    }
+    
+    var windSpeed: String {
+        return "\(city.wind.speed) m/s"
```
Comment (labasbozidar) [inline WeatherPlannerDemo/UI/ViewModels/CityViewModel.swift:28]: You can put "°" and "m/s" into localizations.
I saw that you created MeasurementUnitDescription. This is ok then.

**Review by labasbozidar — CHANGES_REQUESTED** [review]

File: WeatherPlannerDemo/Localization/LocalizationService.swift (hunk 3 above)
Comment (alen3003) [reply WeatherPlannerDemo/Localization/LocalizationService.swift:17]: Will have this in the feature/city_details PR where I already created a file for String extensions

**Review by labasbozidar — APPROVED** [review]

**Pull request merged**